src/object/bonus_block.cpp

```diff
@@ -748,7 +759,7 @@ BonusBlock::preload_contents(int d)
     case 6: // Light.
     case 15: // Light (On).
       SoundManager::current()->preload("sounds/switch.ogg");
-      m_lightsprite=Surface::from_file("/images/objects/lightmap_light/bonusblock_light.png");
+      m_lightsprite = m_sprite->create_linked_sprite("light");
```

Comment:
...What? So `on-light` and `light` are different? And isn't `light` reserved for the light linked sprite? And don't you think the `m_lightsprite` name gets confusing?